---
title: "Test"
author: "Matthieu Tuffery"
date: '2022-03-14'
output: html_document
---

This code makes few thing:
1 )first it imports all DNA sequences from all chromomes from the large SNP dataset
2) it compare the number of difference (and frequency) between two strains for every chromosome
3) it attempt to make a phylogenetic tree using the first chromosome of the first 10 strains 





```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(seqinr) # allows to access read.fasta // you probably have to install it first
library(ape) # alignment
library(muscle) # alignment
library(ggtree) # tree graph
```

S. cerevisiae has 16 chromosomes

```{r}
chromosomes = c("I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX", "X","XI", "XII", "XIII", "XIV", "XV", "XVI")
```

```{r}
all_chromosomes_set = data.frame() # empty dataframe

for (i in 1:16) { # for each chromosome
  #we read the fasta file 
  a_chromosome = read.fasta(paste(
    "doi_10.5061_dryad.hm2jf__v1/SNP-alignment-largeDataset+Spar/SNP-alignment-largeDataset+Spar/chr" # YOU MAY HAVE To ADAPT   THIS PART ACCORDING TO YOUR FILE TOPOLOGY
    ,chromosomes[i],
    "_SNP_scere_largeDataset+Spar.fasta",
    sep = ""), forceDNAtolower = F)
  for (j in 1:10){ # and for the 10 first sequence we add the concatenated sequence as a string in the dataset.
all_chromosomes_set[i,j] = paste(unlist(a_chromosome[j]), sep = "", collapse = "")  }
 
}
 colnames(all_chromosomes_set) = attr(a_chromosome,"name")[1:10]
```


# Same code without readfasta
```{r}
all_chromosomes_set = data.frame() # empty dataframe
number_of_imported_strains = 1:10 # not all strain are always imported / you only need to change this vector to import more/different strains

for (i in 1:3) { # for each chromosome
  
  bob = read.csv(paste(
    "doi_10.5061_dryad.hm2jf__v1/SNP-alignment-largeDataset+Spar/SNP-alignment-largeDataset+Spar/chr" # YOU MAY HAVE To ADAPT   THIS PART ACCORDING TO YOUR FILE TOPOLOGY
    ,chromosomes[i],
    "_SNP_scere_largeDataset+Spar.fasta",
    sep = ""), header = F)
for (j in number_of_imported_strains){
  all_chromosomes_set[i,j] = bob[j*2,1]
}
}
colnames(all_chromosomes_set) = bob[grep(">",bob[,1]),1][number_of_imported_strains]
```


```{r}
# we compare 2 yeast
yeast1 = 1
yeast2 = 4

for (chromosome in 1:2) { # for every chromosome 

  # we make vectors containgthe sequence
seq1 = unlist(strsplit(all_chromosomes_set[chromosome, yeast1], ""))
seq2 = unlist(strsplit(all_chromosomes_set[chromosome, yeast2],""))

# we count the number of differences
count = 0
for (v in 1:length(seq1)) {
  if (!(seq1[v] == seq2[v])) {
    count = count + 1
  } 
}
print(paste(
chromosome,
"  mistakes:",
count,
"  mistake rate:",
(count/length(seq1))*100,
" length:",
length(seq1)
))

}
```

--> the difference rate often varies between 0.5 and 2%. it varies with the strains compared and with the chromosome, 

```{r}

DNA_str_set = DNAStringSet(as.character(all_chromosomes_set[1, number_of_imported_strains])) # we create a DNAstrset form the sequences we have in the dataframe
```


```{r}
Alignment = muscle::muscle(DNA_str_set, quiet = T)# align the sequences
```

```{r}
DNA_bin = as.DNAbin(Alignment) # create a dnabin from the dna_str_set
Dist = dist.dna(DNA_bin, model = "K80") # calculate the distance
Dist_Mat = as.matrix(Dist) # make a distance matrix
colnames(Dist_Mat) = colnames(all_chromosomes_set)[number_of_imported_strains]# add rownames and colnames
rownames(Dist_Mat) = colnames(all_chromosomes_set)[number_of_imported_strains]

Tree<-nj(Dist_Mat)# a tree using neighbourjoining methode
```

```{r}
ggtree(Tree, ignore.negative.edge = T, layout = "circular") +geom_tiplab() # tree
```
very ugly, but it worked. 